docs/introduction/assets.md

```diff
@@ -0,0 +1,130 @@
+
+  incoming.podlets = [response];
+  res.podiumSend(`<div>Hello, Layout</div>`);
+});
+```
+
+### Use a CDN
+
+This approach involves each podlet uploading its assets to a predefined CDN location so that the layout can then include the CDN URLs in its HTML response.
+
+**Step 1.**
+
+In your podlet, upload your assets to a CDN. You might do this whenever your podlet server is built or starts up to ensure the latest version is available on the CDN.
+
+**Step 2.**
+
+Next, tell the podlet the location of your assets so that it can populate the manifest file.
+
+```js
+podlet.js({ value: "http://some-cdn.com/client.js" });
+podlet.css({ value: "http://some-cdn.com/style.css" });
+```
+
+**Step 3.**
+
+Set `incoming.podlets` and use `podiumSend` in your layout's request handler. This way the [document template](api/document.md) can include the CSS and JS assets served by the podlet.
+
+```js
+app.get(layout.pathname(), (req, res) => {
+  const incoming = res.locals.podium;
+  const response = await myPodlet.fetch(incoming);
+
+  incoming.podlets = [response];
+  res.podiumSend(`<div>Hello, Layout</div>`);
+});
+```
+
+## Deduplicating shared dependencies
+
+It's likely one or more of your podlets share a common dependency, such as React. Unless you take action each podlet will bundle its own complete copy of React, wasting bandwith and execution time.
+
+It's up to you to configure your build tools and infrastructure so you can avoid this duplication in your bundles and serve shared dependencies in a performant way.
+
+You may want to look into [Eik](https://eik.dev/docs/overview) and its [build tool plugins](https://eik.dev/docs/mapping_plugins), which were built by the same team that maintains Podium to solve this performance problem.
+
+## Isolation
+
+A podlet should ideally not affect or be affected by the layout or other podlets. This can be tricky, particularly for CSS because of its global nature and the cascade.
+
+### Unique selectors
+
+You can work around the isolation problem by adopting a namespacing convention for all CSS selectors. CSS modules and other similar tools that generate unique selectors can also help mitigate the isolation problem.
+
+Unique selectors can mitigate some of the isolation problems, but a podlet can still be affected by the layout's CSS.
+
+### Declarative shadow DOM
+
+Using the [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM) you can isolate a podlet from its surroundings. By wrapping a podlet in a [declarative shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM#declaratively_with_html) you can still get the benefits of server-side rendering.
+
+#### `podlet.css()` can't be used with Shadow DOM
+
+With `podlet.css()` the end result is a `<link />` tag in the HTML document's `<head />`. If your podlet's content renders inside a shadow DOM that CSS won't be able to reach the podlet.
```

Comment:
```suggestion
With `podlet.css()` the end result is a `<link />` tag in the HTML document's `<head />`. If your podlet's content renders inside a Shadow DOM that CSS won't be able to reach the podlet.
```

Reply:
It's "shadow DOM" on MDN, so I made the title lowercase instead